kanren/cons.py

```diff
@@ -0,0 +1,243 @@
+from functools import reduce
+from itertools import chain
+from operator import length_hint
+from collections import OrderedDict
+from collections.abc import Iterator
+
+from multipledispatch import dispatch
+
+from toolz import drop, last
+
+from unification.core import unify, _unify, reify, _reify
+
+
+def first(x):
+    return next(iter(x), None)
+
+
+def rest(x):
+    return drop(1, x)
+
+
+def none_to_list(x):
+    return [] if x is None else x
+
+
+@dispatch(object, object)
+def cons_merge(car_part, cdr_part):
+    """Merge a generic car and cdr."""
+    return None
+
+
+class ConsType(type):
+    def __instancecheck__(self, o):
+        if is_cons(o):
```

Comment:
Nit: `return is_cons(o)`

Reply:
Looks like a left-over from some refactoring; I'll change it.